schemas/2019-03-01-hybrid/deploymentTemplate.json

```diff
@@ -0,0 +1,3639 @@
+              "enum": [
+                "2018-03-01-hybrid"
+              ]
+            },
+            "resources": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/2019-04-01-profile-resource"
+              }
+            }
+          }
+        },
+        {
+          "properties": {
+            "apiProfile": {
+              "enum": [
+                "2019-03-01-hybrid"
+              ]
+            },
+            "resources": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/2019-03-01-hybrid-profile-resource"
+              }
+            }
+          }
+        },
+        {
+          "properties": {
+            "apiProfile": {
+              "enum": [
+                "2018-06-01-profile"
+              ]
+            },
+            "resources": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/2019-04-01-profile-resource"
+              }
+            }
+          }
+        },
+        {
+          "properties": {
+            "apiProfile": {
+              "enum": [
+                "2017-03-09-profile"
+              ]
+            },
+            "resources": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/2019-04-01-profile-resource"
+              }
+            }
+          }
+        }
+      ]
+    }
+  ],
+  "properties": {
+    "resources": {
+      "type": "array"
+    },
+    "apiProfile": {
+      "type": "string",
+      "enum": [
+        "2018-03-01-hybrid",
+        "2019-03-01-hybrid",
+        "2017-03-09-profile",
+        "2018-06-01-profile"
+      ],
+      "description": "The apiProfile to use for all resources in the template."
+    },
+    "$schema": {
+      "type": "string",
+      "description": "JSON schema reference"
+    },
+    "contentVersion": {
+      "type": "string",
+      "pattern": "(^[0-9]+\\.[0-9]+\\.[0-9]+\\.[0-9]+$)",
+      "description": "A 4 number format for the version number of this template file. For example, 1.0.0.0"
+    },
+    "variables": {
+      "type": "object",
+      "description": "Variable definitions"
+    },
+    "parameters": {
+      "type": "object",
+      "description": "Input parameter definitions",
+      "additionalProperties": {
+        "$ref": "#/definitions/parameter"
+      }
+    },
+    "functions": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/functionNamespace"
+      },
+      "description": "User defined functions"
+    },
+    "outputs": {
+      "type": "object",
+      "description": "Output parameter definitions",
+      "additionalProperties": {
+        "$ref": "#/definitions/output"
+      }
+    }
+  },
+  "additionalProperties": false,
+  "required": [
+    "$schema",
+    "contentVersion",
+    "resources"
+  ],
+  "definitions": {
+    "2019-03-01-hybrid-profile-resource": {
```

Comment:
If you have a script to regenerate this file from the stack-supported list of resources, please could you check this script into this repo?

Reply:
I dont currently have a script... should I make one ?

Reply:
It would certainly be helpful for maintainability for both ARM & Azure Stack going forwards! How did you generate the list of resources? The scripts in ./generator/scripts in this repo are a good starting point and I can give you some other pointers.

Reply:
I did it by hand for adding the resources one by one. To write the script for stack to generate the deployment Template i would need to first grab the api profile comb to see what is supported and then put in the ref to the supported apiprofiles